File: web/src/main/java/by/sorface/passport/web/dao/nosql/redis/RedisOAuth2AuthorizationInitRepository.java
package by.sorface.passport.web.dao.nosql.redis;

import by.sorface.passport.web.dao.nosql.redis.models.RedisOAuth2AuthorizationInit;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.QueryByExampleExecutor;

import java.util.Optional;

public interface RedisOAuth2AuthorizationInitRepository  extends CrudRepository<RedisOAuth2AuthorizationInit, String>,
        QueryByExampleExecutor<RedisOAuth2AuthorizationInit> {

    Optional<RedisOAuth2AuthorizationInit> findFirstByCode(String code);

    Optional<RedisOAuth2AuthorizationInit> findFirstByState(String state);

}

File: web/src/main/kotlin/by/sorface/passport/web/dao/nosql/redis/RedisOAuth2AuthorizationInitRepository.kt
package by.sorface.passport.web.dao.nosql.redis

import by.sorface.passport.web.dao.nosql.redis.models.RedisOAuth2AuthorizationInit
import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.query.QueryByExampleExecutor
import java.util.*

interface RedisOAuth2AuthorizationInitRepository : CrudRepository<RedisOAuth2AuthorizationInit?, String?>, QueryByExampleExecutor<RedisOAuth2AuthorizationInit?> {
    fun findFirstByCode(code: String?): RedisOAuth2AuthorizationInit?

    fun findFirstByState(state: String?): RedisOAuth2AuthorizationInit?
}
